src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -344,17 +345,35 @@ private static bool ReadOutside(ExpressionSyntax dataFlowParent, IdentifierNameS
 
         private static bool FlowsOut(ExpressionSyntax dataFlowParent, ArgumentSyntax decl, IdentifierNameSyntax[] references)
         {
-            foreach (var reference in references)
+            ForStatementSyntax forStatement;
+
+            if ((forStatement = decl.Ancestors().OfType<ForStatementSyntax>().FirstOrDefault()) != null &&
+                 forStatement.Incrementors.Span.Contains(decl.Position) &&
+                 forStatement.Statement.DescendantNodes().OfType<ForStatementSyntax>().Any(f => f.Condition == null))
+            {
+                return false;
+            }
+
+            var containingStatement = decl.Ancestors().OfType<StatementSyntax>().FirstOrDefault();
+            var containingReturnOrThrow = containingStatement as ReturnStatementSyntax ?? (StatementSyntax)(containingStatement as ThrowStatementSyntax);
+
+            MethodDeclarationSyntax methodDeclParent;
+
+            if (containingReturnOrThrow != null && decl.Identifier.ValueText == "x1" && 
```

Comment:
Nevermind, just realized this is in the actual test code.

---
In reply to: [67759021](https://github.com/dotnet/roslyn/pull/12077#discussion_r67759021) [](ancestors = 67759021)